src/System.CommandLine.Subsystems/ValueSources/AggregateValueSource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.ValueSources;
+
+public class AggregateValueSource<T> : ValueSource<T>
+{
+    private List<ValueSource<T>> valueSources = [];
+
+    public AggregateValueSource(ValueSource<T> firstSource,
```

Comment:
It's not clear to me what the description is used for - doc comments would really help here.

I also don't like how we have first, second, other instead of just a single IEnumerable. Plus the splicing of the description midway feels kinda odd.

Reply:
I did some before I ran out of steam. Can we have an analyzer for this?